pages/gall/[id]/index.tsx

```diff
@@ -68,6 +99,19 @@ const Gall = ({ species, imagePaths }: Props): JSX.Element => {
         setSelectedSource(s);
     };
 
+    const sortSourceList = () => {
+        const newSort = sourceSortOptions[(sourceList.sort.id + 1) % sourceSortOptions.length];
```

Comment:
Rather than keep an id field just use an index into the array. Keep the index as React state with `useState` increment each time the sort is changed, resetting to zero when it is congruent to 0 % mod the length of the options.

I am not sure I made sense. If not just ask!